packages/components/src/Filters/FilterDropdown.tsx

```diff
@@ -0,0 +1,94 @@
+import React, { useState } from 'react';
+import { getDefaultOUIAId } from '@patternfly/react-core/';
+import { Dropdown, DropdownToggle, Level } from '@patternfly/react-core';
+import FilterInput from './FilterInput';
+import './filter-dropdown.scss';
+
+export type FilterCategory = {
+  title?: string;
+  type?: string;
+  urlParam?: string;
+  values?: {
+    label?: string;
+    value?: string;
+  }[];
+};
+
+export interface FilterDropdownProps {
+  addFilter?: (filterName?: string, selectedValue?: string | number, type?: string) => void;
+  removeFilter?: (filterName?: string, selectedValue?: string | number) => void;
+  hideCategories?: [string | undefined];
+  filters?: Record<string, string>;
+  filterCategories?: FilterCategory[];
+  label?: React.ReactNode;
+  ouiaId?: string;
+  ouiaSafe?: boolean;
+}
+
+const ouiaStateId = getDefaultOUIAId('RHI/FilterDropdown');
+
+const FilterDropdown: React.FunctionComponent<FilterDropdownProps> = ({
+  hideCategories,
+  filters,
+  filterCategories,
+  label,
+  ouiaId,
+  ouiaSafe = true,
+  addFilter,
+  removeFilter,
+  ...props
+}) => {
+  const [isOpen, setIsOpen] = useState(false);
+
+  const addRemoveFilters = (selectedValue?: string | number, filterName?: string, type?: string, isChecked?: boolean) => {
+    if (type === 'checkbox') {
+      isChecked ? addFilter?.(filterName, selectedValue, type) : removeFilter?.(filterName, selectedValue);
+    } else {
+      addFilter?.(filterName, selectedValue, type);
+    }
+  };
+
+  const ouiaFinalId = ouiaId !== undefined ? ouiaId : ouiaStateId;
```

Comment:
Good thing moving `ouiaStateId` from the component! We can further improve this by assigning default value on line 35
```JS
ouiaId = ouiaStateId,
```

It's the same, but it'll make the component slightly cleaner.